migrations/tenant/0029-object-bucket-name-sorting.sql

```diff
@@ -0,0 +1,2 @@
+-- postgres-migrations disable-transaction
+CREATE UNIQUE INDEX CONCURRENTLY IF NOT EXISTS idx_name_bucket_unique on storage.objects (name COLLATE "C", bucket_id);
```

Comment:
For the purpose of migration I think the index can be created without `COLLATE "C"`, because the `LIKE` operator is not used.
But it shouldn't affect anything in terms of performance or functionality, so it is up to you.

Reply:
I think it matter that we use COLLATE "C" in this index, since we order the results by name COLLATE "C"